### PR TITLE
Added search parameters to qdrant max_marginal_relevance_search

### PR DESCRIPTION
**Pull request by whitead**
Adds the qdrant search filter/params to the `max_marginal_relevance_search` method, which is present on others. I did not add `offset` for pagination, because it's behavior would be ambiguous in this setting (since we fetch extra and down-select). 

@rlancemartin, @eyurtsev

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/qdrant.py [hunk 1]
```diff
@@ -712,6 +716,25 @@ def max_marginal_relevance_search(
...
+                - int - number of replicas to query, values should present in all
+                        queried replicas
+                - 'majority' - query all replicas, but return values present in the
+                               majority of replicas
+                - 'quorum' - query the majority of replicas, return values present in
+                             all of them
+                - 'all' - query all replicas, and return values present in all replicas
         Returns:
             List of Documents selected by maximal marginal relevance.
         """
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/qdrant.py:740]: don't we need to pass all the new params through in line 720

Comment (whitead) [reply libs/langchain/langchain/vectorstores/qdrant.py:740]: Thanks for looking over this! Those are possible values for the consistency flag, not additional parameters.

Comment (baskaryan) [reply libs/langchain/langchain/vectorstores/qdrant.py:740]: i meant shouldn't L720 (in original code)

```python
        return self.max_marginal_relevance_search_by_vector(
            query_embedding, k, fetch_k, lambda_mult, **kwargs
        )
```

be

```python
        return self.max_marginal_relevance_search_by_vector(
            query_embedding, k, fetch_k, lambda_mult, filter=filter, search_params=search_params, score_threshold=score_threshold, consistency=consistency, **kwargs
        )
```

and then these params also need to be added to signature of max_marginal_relevance_search_by_vector (L813 in original code)

File: libs/langchain/langchain/vectorstores/qdrant.py [hunk 2]
```diff
@@ -841,9 +864,14 @@ def max_marginal_relevance_search_with_score_by_vector(
         results = self.client.search(
             collection_name=self.collection_name,
             query_vector=query_vector,
+            query_filter=filter,
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/qdrant.py:867]: re above, filter's not defined anywhere atm

**Comment by whitead** [comment]
Hi @baskaryan, regarding your comments: this PR is 2 weeks old and this file got rewritten after you merged in main. You can see the [original diff]( https://github.com/langchain-ai/langchain/pull/7745/commits/b183f3107d866805b815ebf1614e88ffc2205e94). I will revise it to support the new async rewrite which refactored the method I changed and ping you once that's done.

**Comment by baskaryan** [comment]
> Hi @baskaryan, regarding your comments: this PR is 2 weeks old and this file got rewritten after you merged in main. You can see the [original diff](https://github.com/langchain-ai/langchain/pull/7745/commits/b183f3107d866805b815ebf1614e88ffc2205e94). I will revise it to support the new async rewrite which refactored the method I changed and ping you once that's done.

ah makes a lot of sense, had my suspicions. can also take over and land if you'd like!

**Comment by baskaryan** [comment]
@kacperlukawski would love your eyes on this one

**Comment by whitead** [comment]
@kacperlukawski @baskaryan -> added unit test for my use-case. Please let me know if there are any other blockers to merging.

**Comment by whitead** [comment]
@kacperlukawski @baskaryan bumping this again, thanks

**Comment by kacperlukawski** [comment]
@whitead I'm on it. I tried running the test, but it failed.

**Comment by kacperlukawski** [comment]
@whitead I started a PR in your fork, so all the tests pass.

**Comment by whitead** [comment]
Thanks @kacperlukawski! Merged

**Comment by whitead** [comment]
@kacperlukawski pinging again (sorry to be annoying, but we want to get our infra back on latest langchain!)

**Comment by baskaryan** [comment]
passing integration tests so i think safe to merge. thanks a bunch @whitead and @kacperlukawski!!

**Pull request merged**